src/OmniSharp.Roslyn/OmniSharpWorkspace.cs

```diff
@@ -80,9 +83,53 @@ public void RemoveMetadataReference(ProjectId projectId, MetadataReference metad
 
         public void AddDocument(DocumentInfo documentInfo)
         {
+            var documentId = GetDocumentId(documentInfo.FilePath);
```

Comment:
Can 86-93 be a call to TryRemoveMiscellaneousDocument(documentInfo.filePath)?  Looks same functionality wise

Reply:
@SirIntruder Done. Thanks!